datafusion/physical-plan/src/filter.rs

```diff
@@ -379,7 +430,8 @@ impl Stream for FilterExecStream {
             match ready!(self.input.poll_next_unpin(cx)) {
                 Some(Ok(batch)) => {
                     let timer = self.baseline_metrics.elapsed_compute().timer();
-                    let filtered_batch = batch_filter(&batch, &self.predicate)?;
+                    let filtered_batch =
```

Comment:
I recommend adding a test so we don't accidentally break the feature by accident